banyan/banyan/sessions.py

```diff
@@ -1,8 +1,194 @@
+import io
+import json
+import os
+import shutil
+import time
+import zipfile
+
+import boto3
+from botocore.exceptions import ClientError
+
+from banyan.config import configure
+from banyan.utils import convert_iso_time, send_request_get_response
+
+
 SessionId = str
+SUPPORTED_PYTHON_VERSIONS = ["3.7", "3.8", "3.9"]
+
+_iam_client = boto3.client("iam")
+_lambda_client = boto3.client("lambda")
+_s3_client = boto3.client("s3")
+
+
+def _get_executor_code_from_s3():
+    return _s3_client.get_object(
+        Bucket="banyan-executor",
+        Key="executor.py"
+    )
+
+
+def _get_executor_lambda_zip(site_packages_dir: str):
+    # Construct zipfile with site_packages_dir if provided
+    zipfile_name = "executor_lambda_code"
+    if site_packages_dir:
+        shutil.make_archive(zipfile_name, "zip", root_dir=site_packages_dir, base_dir=".")
+    else:
+        # Create empty zipfile
+        with zipfile.ZipFile(zipfile_name + ".zip", "w") as f:
+            pass
+    # Add executor code file
+    executor_code = _get_executor_code_from_s3()["Body"].read().decode("utf-8")
+    with zipfile.ZipFile(zipfile_name + ".zip", "a") as f:
+        f.writestr("lambda_function.py", executor_code)
+    # Upload to S3. First create a S3 bucket if not already existing
+    # and then upload the zip file.
+    bucket_name = "banyan-assets"
+    s3_key = zipfile_name + ".zip"
+    try:
+        _s3_client.create_bucket(
+            Bucket=bucket_name,
+            CreateBucketConfiguration={"LocationConstraint": _s3_client.meta.region_name}
+        )
+        _s3_client.get_waiter("bucket_exists").wait(Bucket=bucket_name)
+    except ClientError as error:
+        if not error.response["Error"]["Code"] in ["BucketAlreadyExists", "BucketAlreadyOwnedByYou"]:
+            raise
```

Comment:
Does this raise `error`? If so that's good

Reply:
Yes, it will re-raise the original error.